### PR TITLE
Repurpose project as hello world starter app for developer center

### PR DESCRIPTION
**Pull request by tstapleton**
Almost everything that existed before is now gone. 🔥

Two commands help getting started with app development:

* `npm run create-app` - creates an app and releases it on Tradeshift, writes manifest.json
* `npm run start-app` - starts the app along with a tunnel (using [localtunnel.me](https://localtunnel.github.io/www/)) so requests for the app are redirected to the local server

Everything for those commands are under `/scripts` so it is separated from the app source code.

And there is a very simple node server under `/src` that will handle requests for the app.

~The icon didn't work during the demo just now - I'll push a fix when I have it.~ Fixed and pushed.

@Tradeshift/dx

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,39 +1,48 @@
...
-    "node-rest-client": "^1.5.1",
-    "node-uuid": "^1.4.7",
-    "path": "^0.12.7",
-    "q": "^1.4.1",
-    "superagent": "^1.6.1"
-  }
+	"name": "bucko",
+	"author": "Tradeshift",
+	"version": "0.0.2",
+	"description": "A sample app for getting started developing for the Tradeshift platform",
```
Comment (TimeBomb) [inline package.json]: `for the Tradeshift platform` -> `on the Tradeshift platform` maybe?

File: scripts/create-app.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+require('dotenv').config();
+
+const api = require('./helpers/ts-api');
+const chalk = require('chalk');
+const templates = require('./helpers/templates');
+const utils = require('./helpers/utils');
+
+const TUNNEL_DOMAIN = 'localtunnel.me';
+
+const logSuccess = ({ appName, appEmbedUrl, appStoreUrl }) => {
```
Comment (TimeBomb) [inline scripts/create-app.js]: Don't think I've seen this syntax before, what's putting the curly braces around the arguments do in this context? Is it expecting a single object and dereferencing these three object properties so they can be used as normal variables in our function?

File: scripts/helpers/templates.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+			<path
+				d="M130.56 40.96c-1.452-1.447-3.651-2.124-6.922-2.124-6.55 0-19.394 3.092-27.923 11.626L78.019 68.154h-29.82a4.502 4.502 0 0 0-3.171 1.312l-3.168 3.172a4.489 4.489 0 0 0 0 6.34l50.687 50.682a4.468 4.468 0 0 0 3.168 1.312 4.468 4.468 0 0 0 3.167-1.312l3.172-3.168a4.474 4.474 0 0 0 1.308-3.172V93.497l17.696-17.696c6.442-6.438 9.282-14.753 10.398-19.125.905-3.553 2.63-12.19-.896-15.716zM69.063 77.114l-8.194 8.194-8.198-8.194h16.392zm-1.86 14.533l31.679-31.678 12.67 12.665-31.674 31.682-12.674-12.67zm27.203 27.198l-8.194-8.19 8.194-8.198v16.388zm23.166-52.86L105.53 53.943c6.272-4.404 14.036-6.147 18.108-6.147h.126c.054 3.611-1.626 11.684-6.192 18.19zM67.2 116.99c-2.017 2.012-9.709 3.754-17.267 4.592.856-7.585 2.63-15.304 4.597-17.266 1.156-1.16 2.567-1.895 4.045-2.29l-6.764-6.764c-1.282.753-2.518 1.617-3.616 2.715-6.15 6.155-7.441 24.631-7.638 28.273l-.269 4.982 4.986-.269c3.642-.197 22.118-1.496 28.265-7.643 1.102-1.102 1.962-2.334 2.714-3.62l-6.764-6.769c-.39 1.492-1.125 2.894-2.29 4.06z"
+				fill-rule="nonzero"
+				fill="#FFF" />
+		</g>
+	</svg>
+`);
+
+const appVersion = ({ appId, appName = appId, appUrl, vendorId, version = '0.0.1' }) => ({
```
Comment (TimeBomb) [inline scripts/helpers/templates.js]: This app is to help new 3rd party TS devs, right? May be nice to link to documentation that helps some of this object, e.g. what are the different access capabilities?

File: scripts/helpers/ts-api.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	}
+}
+
+const doRequest = async (auth, { body, method = 'GET', url }) => {
+	const options = {
+		body,
+		gzip: true,
+		headers: {
+			'user-agent': 'bucko-app',
+			'x-tradeshift-tenantid': auth.companyId,
```
Comment (TimeBomb) [inline scripts/helpers/ts-api.js]: I normally see this as something like `X-Tradeshift-TenantID`, does capitalization matter?

File: scripts/helpers/utils.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+const writeFileAsync = promisify(fs.writeFile);
+
+const buildAppStoreUrl = ({ appId, tsApiHost, vendorId }) =>
+	`${tsApiHost.replace('api-', '')}/#/apps/Tradeshift.AppStore/apps/${vendorId}.${appId}`;
+
+const buildAppEmbedUrl = ({ appId, tsApiHost, vendorId }) =>
+	`${tsApiHost.replace('api-', '')}/#/apps/${vendorId}.${appId}`;
+
+const buildAppUrl = ({ appId, host, vendorId }) =>
+	`https://${vendorId.toLowerCase()}${appId.toLowerCase()}.${host}/`;
```
Comment (TimeBomb) [inline scripts/helpers/utils.js]: Would the ever be a case where the protocol is not `https`?

File: src/app.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+const app = new Koa();
+const router = new Router();
+
+// nicer error pages for development
+app.use(error());
+
+// create session - needed by passport for authentication
+// random key used so oauth dance is performed every time
+// TODO: implement refresh token strategy and use a constant key
```
Comment (TimeBomb) [inline src/app.js]: leaving TODO comments in; is the implementation we have here good enough for now?

File: package.json (hunk 1 above)
Comment (tstapleton) [reply package.json]: yeah

File: scripts/helpers/templates.js (hunk 3 above)
Comment (tstapleton) [reply scripts/helpers/templates.js]: the stuff in scripts/* isn't for 3rd party devs - the starter app is in src/*. the stuff in scripts/* is probably better suited to be included as a package dependency, but will do that later, if ever.

File: scripts/helpers/ts-api.js (hunk 4 above)
Comment (tstapleton) [reply scripts/helpers/ts-api.js]: headers are case insensitive per the http spec - https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html

File: scripts/helpers/utils.js (hunk 5 above)
Comment (tstapleton) [reply scripts/helpers/utils.js]: tradeshift requires iframed apps to be hosted on https...hence all of the ngrok/localtunnel stuff.

File: src/app.js (hunk 6 above)
Comment (tstapleton) [reply src/app.js]: yeah, it needs to be

**Comment by tstapleton** [comment]
> Do we want the app to be developable on Windows? Can double check the npm run scripts.

@TimeBomb yeah, good idea. i think it can wait until after the demos though.

**Pull request merged**